DisCatSharp/Net/Rest/DiscordApiClient.cs

```diff
@@ -1200,6 +1199,307 @@ internal async Task UpdateUserVoiceStateAsync(ulong guild_id, ulong user_id, ulo
         }
         #endregion
 
+        #region Guild Scheduled Events
+
+        /// <summary>
+        /// Creates a scheduled event.
+        /// </summary>
+        internal async Task<DiscordScheduledEvent> CreateGuildScheduledEventAsync(ulong guild_id, Optional<ulong?> channel_id, Optional<DiscordScheduledEventEntityMetadata> metadata, string name, ScheduledEventPrivacyLevel privacy_level, DateTimeOffset scheduled_start_time, Optional<DateTimeOffset?> scheduled_end_time, Optional<string> description, ScheduledEventEntityType type, string reason = null)
+        {
+            var pld = new RestGuildScheduledEventCreatePayload
+            {
+                ChannelId = channel_id,
+                EntityMetadata = metadata,
+                Name = name,
+                PrivacyLevel = privacy_level,
+                ScheduledStartTime = scheduled_start_time,
+                ScheduledEndTime = scheduled_end_time,
+                Description = description,
+                EntityType = type
+            };
+
+            var headers = Utilities.GetBaseHeaders();
+            if (!string.IsNullOrWhiteSpace(reason))
+                headers[REASON_HEADER_NAME] = reason;
+
+            var route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.SCHEDULED_EVENTS}";
+            var bucket = this.Rest.GetBucket(RestRequestMethod.POST, route, new { guild_id }, out var path);
+
+            var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+            var res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.POST, route, headers, DiscordJson.SerializeObject(pld));
+
+            var scheduled_event = JsonConvert.DeserializeObject<DiscordScheduledEvent>(res.Response);
+            var guild = this.Discord.Guilds[guild_id];
+
+            scheduled_event.Discord = this.Discord;
+            guild._scheduledEvents.TryAdd(scheduled_event.Id, scheduled_event);
```

Comment:
This also feels asinine? Let the callee handle this, imo.

Reply:
Also you're forgetting to initialize the metadata and set a client on the creator object.

Reply:
Oh :/